_commands/tools/grep.md

```diff
@@ -3,22 +3,100 @@
 ~~~
 
-<!--more-->
+By default, `grep` will print all the matching lines to standard output (the console) with the matching parts colored. This behavior can be modified by using one or more of the the command line options `grep` offers.
 
 ### Useful Options / Examples
 
-#### Example command
+#### -c, -&#45;count
+
+##### Using the `-c` or `--count` option overrides the default output behavior of `grep`. Instead of printing all of the matches to the console, it will print the number of lines that match. This is a useful option if, for example, you only want to know if there are any matches (count &lt;&gt; 0) but aren't as concerned with what those matches actually are.
+
+~~~ bash
+$ grep -c "wolf" animals.txt        // counts all animals containing "wolf"
+187
+$ grep -c "^ll.*" animals.txt       // counts all animals that begin with "ll"
+1
+$ grep -c "axb$" animals.txt        // counts all animals that end with "axb"
+0
+~~~
+
+#### -v, -&#45;invert-match
+
+##### Regular expressions can be tricky to use, especially when you want to find text that doesn't match a particular regex. This is made easier by the `-v` or `--invert-match` option, which finds lines in the target input that do not match the given regular expression.
+
+~~~ bash
+$ grep -v "^[aeiou]" animals.txt
+// finds and prints all animals that do not begin with a lower-case vowel
+$ grep -c -v "s" animals.txt        // counts all animals that don't contain an "s"
+17460
+~~~
+
+#### -i, -&#45;ignore-case
+
+##### Regular expressions, by their nature, are case sensitive. Sometimes, it can be annoying to construct a regex that inherently ignores case differences between letters; this can be especially true when working with extended ASCII or Unicode characters. The `-i` or `--ignore-case` option handles this automatically.
+
+~~~ bash
+$ grep -i "^A" animals.txt
+// finds and prints all animals that begin with an "a" or an "A"
+~~~
+
+#### Other Useful Options
+
+##### `-n`, `--line-number` will list the line number of the matching line before the contents
+`-l`, `--files-with-matches` will list the input files that contain a matching line, not the matching lines themselves
+
+### Input/Output
+
+##### `grep` can be used with one file or with multiple files by listing more than one in the input position for the command. You can even use regular expressions to match multiple files!
+
+~~~ bash
+$ grep "umich" file1.txt file2.txt       // search two files explicitly
+// output here
+$ grep "umich" *.txt                 // search all .txt files in the current directory
+// output here
+$ grep "umich" .                         // search all files in the current directory
+~~~
 
-##### Break it down
+Input can also be piped into `grep` from another program or redirected using &lt;. Likewise, the output of `grep` can be piped into another program (including another `grep`) or redirected using &gt;
 
-#### Example command
+### Regular Expression Basics
```

Comment:
For this section, it would be helpful to link to a regex guide so people can learn more on their own. For example, where did you get this information from? Linking or explaining that would be good.

Also, if you can apply the `code` formatting to the characters that require it. Adding that makes it easier to read.
For example, the '^A' in your first example should be `^A`

Reply:
I applied the code formatting to all the characters, both in the example sub-bullets and in the primary bullets. Also linked to three resources at the bottom.